test/blackbox/run_blackbox_tests/test_cases_encrypted_storage.c

```diff
@@ -0,0 +1,764 @@
+    3. Reopen both node instances and obtain node handles.
+
+    Expected Result:
+    When NUT and peer node is restarted it should obtain node handles appropriate node handles.
+    (i.e, node handles of itself and other node's handle)
+*/
+static bool test_steps_encrypted_storage_04(void) {
+
+	char key[] = "encryption_key";
+	size_t keylen = sizeof(key);
+
+	// Open NUT and peer node instances
+
+	meshlink_set_log_cb(NULL, MESHLINK_DEBUG, logger);
+	meshlink_handle_t *mesh1 = meshlink_open_encrypted("encrypted_conf.1" , "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	meshlink_handle_t *mesh2 = meshlink_open_encrypted("encrypted_conf.2" , "peer", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert(mesh1);
+	assert(mesh2);
+
+	meshlink_set_log_cb(mesh1, MESHLINK_DEBUG, logger);
+	meshlink_set_log_cb(mesh2, MESHLINK_DEBUG, logger);
+
+	// Import and Export key's and addresses of each other
+
+	char *export1 = meshlink_export(mesh1);
+	char *export2 = meshlink_export(mesh2);
+	assert_int_not_equal(export1, NULL);
+	assert_int_not_equal(export2, NULL);
+
+	assert_int_equal(meshlink_import(mesh1, export2), true);
+	assert_int_equal(meshlink_import(mesh2, export1), true);
+
+	assert(meshlink_get_node(mesh1, "peer"));
+	assert(meshlink_get_node(mesh2, "nut"));
+
+	// Close NUT and peer node instances
+
+	meshlink_close(mesh1);
+	meshlink_close(mesh2);
+
+	// Re-open NUT and peer node instances
+
+	mesh1 = meshlink_open_encrypted("encrypted_conf.1" , "nut", "test", DEV_CLASS_BACKBONE, key, keylen);
+	mesh2 = meshlink_open_encrypted("encrypted_conf.2" , "peer", "test", DEV_CLASS_BACKBONE, key, keylen);
+	assert_int_not_equal(mesh1, NULL);
+	assert_int_not_equal(mesh2, NULL);
+	meshlink_set_log_cb(mesh1, MESHLINK_DEBUG, logger);
+	meshlink_set_log_cb(mesh2, MESHLINK_DEBUG, logger);
+
+	// Obtain node handles and assert on the result
+
+	assert_int_not_equal(meshlink_get_node(mesh1, "nut"), NULL);
+	assert_int_not_equal(meshlink_get_node(mesh1, "peer"), NULL);
+	assert_int_not_equal(meshlink_get_node(mesh2, "nut"), NULL);
+	assert_int_not_equal(meshlink_get_node(mesh2, "peer"), NULL);
+
+	// Cleanup
+
+	meshlink_close(mesh1);
+	meshlink_close(mesh2);
+
+	meshlink_destroy("encrypted_conf.1");
+	meshlink_destroy("encrypted_conf.2");
+	return true;
+}
+
+/* Execute encrypted storage Test Case # 5 */
+static void test_case_encrypted_storage_05(void **state) {
+	execute_test(test_steps_encrypted_storage_05, state);
+}
+
+/* Test Steps for encrypted storage Test Case # 5
+
+    Test Steps:
+    1. Open & join NUT and peer node instance with encrypted confbase with
+        peer node inviting NUT node.
+    2. Reopen both node instances.
+
+    Expected Result:
+    When NUT and peer node is restarted it should obtain node handles appropriate node handles
+    (i.e, node handles of itself and other node's handle) and also should see other node's reachable
+    status from callbacks.
```

Comment:
This test includes all the checks that test case #4 does. So I think test case #4 is redundant.

Reply:
So just to confirm invite and join API's and literally have the same output like we have with import and export API? Export and Importing other than exchanging addresses and public key do they exchange anything else?

Reply:
Ah, if you indeed want to explicitly test both import/export and invite/join separately, then keep the two tests. But then I would also check that the NUT and peer are reachable after the join in test case #4. To confirm, normally the end result of invite/join should be the same as import/export.